### PR TITLE
feat: add a parameter in circle producer to filter on user's own pipe…

### PR DESCRIPTION
**Pull request by cpoissonnier**
As promised, here is the pull request for the [feature change I asked](https://github.com/redaxmedia/chroma-feedback/issues/85)

### CODE REVIEW

**Comment by cpoissonnier** [comment]
Sorry about the failing tests, I could not run them on my local environment 😕

I don't really understand why they fail. Is it linked to what I developed or is it about missing properties ?

File: chroma_feedback/producer/circle/core.py [hunk 1]
```diff
@@ -11,6 +12,7 @@ def init(program : ArgumentParser) -> None:
 
 	if not ARGS:
 		program.add_argument('--circle-host', default = 'https://circleci.com')
+		program.add_argument('--circle-only-mine', default = 'false')
```
Comment (redaxmedia) [inline chroma_feedback/producer/circle/core.py:15]: Use this instead:

program.add_argument('--circle-only-mine', action = 'store_true'))

File: chroma_feedback/producer/circle/core.py [hunk 2]
```diff
@@ -22,17 +24,23 @@ def run() -> List[Dict[str, Any]]:
...
 	return result
 
 
-def fetch(host : str, organization : str, slug : str, token : str) -> List[Dict[str, Any]]:
+def fetch(host : str, organization : str, slug : str, token : str, only_mine : str) -> List[Dict[str, Any]]:
 	result = []
 	response = None
 
-	if host and slug and token:
+	if only_mine and bool(strtobool(only_mine)) and host and slug and token:
```
Comment (redaxmedia) [inline chroma_feedback/producer/circle/core.py:37]: Remove the type casting and move only_mine to the end according to the function parrams

File: chroma_feedback/producer/circle/core.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+from distutils.util import strtobool
```
Comment (redaxmedia) [inline chroma_feedback/producer/circle/core.py:1]: Remove that once you got rid of type casting

File: tests/producer/circle/test_core.py [hunk 4]
```diff
@@ -3,9 +3,20 @@
...
+		assert result[0]['slug'] == 'gh/redaxmedia/chroma-feedback/lint-and-test'
+		assert result[0]['active'] is True
+		assert result[0]['status']
+	else:
+		pytest.skip('CIRCLE_TOKEN is not defined')
+
 def test_fetch_slug() -> None:
 	if 'CIRCLE_TOKEN' in os.environ:
-		result = fetch('https://circleci.com', None, 'github/redaxmedia/chroma-feedback', os.environ['CIRCLE_TOKEN'])
+		result = fetch('https://circleci.com', None, 'github/redaxmedia/chroma-feedback', os.environ['CIRCLE_TOKEN'], 'false')
```
Comment (redaxmedia) [inline tests/producer/circle/test_core.py:19]: It is `False` in python...

File: tests/producer/circle/test_core.py [hunk 5]
```diff
@@ -17,7 +28,7 @@ def test_fetch_slug() -> None:
 
 def test_fetch_organization() -> None:
 	if 'CIRCLE_TOKEN' in os.environ:
-		result = fetch('https://circleci.com', 'github/redaxmedia', None, os.environ['CIRCLE_TOKEN'])
+		result = fetch('https://circleci.com', 'github/redaxmedia', None, os.environ['CIRCLE_TOKEN'], 'false')
```
Comment (redaxmedia) [inline tests/producer/circle/test_core.py:31]: It is `False` in python...

File: tests/producer/circle/test_core.py [hunk 6]
```diff
@@ -3,9 +3,20 @@
 from chroma_feedback.producer.circle.core import fetch
 
 
+def test_fetch_only_mine() -> None:
```
Comment (redaxmedia) [inline tests/producer/circle/test_core.py:6]: Please rename to `test_fetch_slug_mine_only` and change order after `test_fetch_slug`

File: tests/producer/circle/test_core.py [hunk 7]
```diff
@@ -3,9 +3,20 @@
...
+	if 'CIRCLE_TOKEN' in os.environ:
+		result = fetch('https://circleci.com', None, 'github/redaxmedia/chroma-feedback', os.environ['CIRCLE_TOKEN'], 'true')
```
Comment (redaxmedia) [inline tests/producer/circle/test_core.py:8]: Please use `True`...

File: chroma_feedback/producer/circle/core.py [hunk 8]
```diff
@@ -22,17 +24,23 @@ def run() -> List[Dict[str, Any]]:
...
-			result.extend(fetch(ARGS.circle_host, None, slug, ARGS.circle_token))
+			result.extend(fetch(ARGS.circle_host, None, slug, ARGS.circle_token, ARGS.circle_only_mine))
 	elif ARGS.circle_organization:
-		result.extend(fetch(ARGS.circle_host, ARGS.circle_organization, None, ARGS.circle_token))
+		result.extend(fetch(ARGS.circle_host, ARGS.circle_organization, None, ARGS.circle_token, ARGS.circle_only_mine))
 	return result
 
 
-def fetch(host : str, organization : str, slug : str, token : str) -> List[Dict[str, Any]]:
+def fetch(host : str, organization : str, slug : str, token : str, only_mine : str) -> List[Dict[str, Any]]:
```
Comment (redaxmedia) [inline chroma_feedback/producer/circle/core.py:33]: Token should be the last parameter ... it is a bit related to slug so I found this better

File: README.md [hunk 9]
```diff
@@ -154,12 +154,13 @@ chroma-feedback --producer=bitbucket
...
-| Host         | https://circleci.com | optional  |
-| Slug         |                      | optional  |
-| Organization |                      | optional  |
-| Token        |                      | required  |
+| Name             | Default              | Mandatory |
+|------------------|----------------------|-----------|
+| Host             | https://circleci.com | optional  |
+| Slug             |                      | optional  |
+| Organization     |                      | optional  |
+| Token            |                      | required  |
```
Comment (redaxmedia) [inline README.md:162]: Change the order... Token at the end

File: README.md [hunk 10]
```diff
@@ -179,6 +180,15 @@ chroma-feedback --producer=circle
 --circle-token <token>
 ```
 
+Monitor my own pipelines (won't work for a whole organisation):
```
Comment (redaxmedia) [inline README.md:183]: Please use:

Monitor a single build with only mine flag:

```
chroma-feedback --producer=circle
--circle-slug <username/repository>
--circle-only-mine
--circle-token <token>
```

**Comment by redaxmedia** [comment]
> Sorry about the failing tests, I could not run them on my local environment confused
> I don't really understand why they fail. Is it linked to what I developed or is it about missing properties ?

Well, the problem is that for security reason the ENV seemed to be blocked in PR cause someone could print them in the logs. The unit testing suite does not test against empty ENV and this has been fixed in the master.

**Comment by redaxmedia** [comment]
I'll fix the review on my own - don't mind about it anymore...

**Pull request merged**